src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +237,55 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+int32_t SystemNative_GetUName(uint32_t uid, char* buffer, size_t bufferSize)
+{
+    assert(buffer != NULL);
+    assert(bufferSize > 0);
+
+    if (buffer == NULL || bufferSize <= 0)
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    struct passwd u;
+    struct passwd* pu; 
+
+    errno = 0;
+    int e;
+    while ((e = getpwuid_r(uid, &u, buffer, bufferSize, &pu)) == EINTR);
+
+    if (e < 0)
+    {
+        errno = e;
+        return -1;
+    }
+    return (int32_t)strlen(buffer);
+}
+
+int32_t SystemNative_GetGName(uint32_t gid, char* buffer, size_t bufferSize)
+{
+    assert(buffer != NULL);
+    assert(bufferSize > 0);
+
+    if (buffer == NULL || bufferSize <= 0)
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    struct group g;
+    struct group* pg; 
+
+    errno = 0;
+    int e;
+    while ((e = getgrgid_r(gid, &g, buffer, bufferSize, &pg)) == EINTR);
+
+    if (e < 0)
+    {
+        errno = e;
+        return -1;
+    }
+    return (int32_t)strlen(buffer);
```

Comment:
According the [docs](https://linux.die.net/man/3/getpwuid_r):

> store the retrieved passwd structure in the space pointed to by pwd. The string fields pointed to by the members of the passwd structure are stored in the buffer buf of size buflen.

What we have here right now works because `pw_name` is the first field of the `passwd` structure. I am OK with that as it does not require an extra copy (perf 👍), but just to be extra safe we should imo add an assert that verifies that the content of `u.pw_name` equals to the content of the buffer.

It could be sth like:


```suggestion
    assert(strcmp(u.pw_name, buffer)== 0);
    return (int32_t)strlen(buffer);
```

Reply:
I made a similar comment: https://github.com/dotnet/runtime/pull/69107#discussion_r871086917.

I strongly dislike the using this buffer directly because it is not the public API.
Performing the copy won't have a measurable impact.

Reply:
> I made a similar comment:

It looks like I started reviewing before you have posted your review and hence duplicated a lot of comments. 

I agree that we should most likely use the public API. This would just always work and be more obvious.

Reply:
Alright, no problem. I'll copy the string.